src/core/engine/processors/mongo_order_status_history_distribution.py

```diff
@@ -0,0 +1,59 @@
+from abc import ABCMeta
+
+from src.constants.etl_status import Service
+
+from src.core.engine.processors.abstract_processor import AbstractProcessor
+from src.lib.repositories.impl_no_sql.order_status_history_repository_impl import (
+    OrderStatusHistoryRepositoryImpl as MongoOrderStatusHistoryRepository,
+)
+
+
+class MongoOrderStatusHistoryDistribution(AbstractProcessor, metaclass=ABCMeta):
+    def __init__(self, app_processor_config, app_context=None):
+        super().__init__(
+            app_processor_config=app_processor_config, app_context=app_context
+        )
+        self.mongo_order_status_history_repository = None
+
+    def set_app_context(self, app_context):
+        ioc = app_context.ioc
+        self.mongo_order_status_history_repository = ioc.get(
+            MongoOrderStatusHistoryRepository
+        )
+        self.app_context = app_context
+
+    def process(self, delta_time):
+        self.mongo_order_status_history_distribution()
+
+    def mongo_order_status_history_distribution(self):
```

Comment:
Logic for both etl seems to be duplicated, could you refactor this function?

Reply:
Done.